test/dates/io.jl

```diff
@@ -359,3 +417,38 @@ let
     @test DateTime(ds, format) == dt
     @test DateTime(ds, escaped_format) == dt
 end
+
+# PR: 15588
+let dt = Dates.DateTime(1991, 2, 3, 4, 5, 6, 7)
+    @test Dates.format(dt, "y") == "1991"
+    @test Dates.format(dt, "yy") == "91"
+    @test Dates.format(dt, "yyyy") == "1991"
+    @test Dates.format(dt, "yyyyyy") == "001991"
+    @test Dates.format(dt, "m") == "2"
+    @test Dates.format(dt, "mm") == "02"
+    @test Dates.format(dt, "mmm") == "002"
+
+    @test Dates.format(dt, "s") == "7"
+    @test Dates.format(dt, "ssss") == "0007"
+    @test Dates.format(dt, ".s") == ".007"
+    @test Dates.format(dt, ".ssss") == ".0070"
+
```

Comment:
unneeded

otherwise looks okay to me, any performance implications? is there a set of dates benchmarks in the suite that nanosoldier runs yet?